sunpy/coordinates/tests/test_frameattributes.py

```diff
@@ -116,6 +116,17 @@ def test_observer_not_hgs_astropy(oca):
     assert converted
 
 
+def test_observer_not_default_representation(oca):
```

Comment:
Am I right in thinking that this doesn't actually run any code that would have caused the bug beforehand? If so it would be good to add a test for a workflow a user would use that would have errored before this fix.

Reply:
It does, it just directly calls the converter function, which is automatically called when assigning an observer attribute to a sunpy frame. (Via a lot of descriptors in astropy)

Reply:
As @Cadair says, the test I wrote does fail on bugged code, but it's not how a user normally interfaces with the `observer` frame attribute.  I've added another test that is more like user code.